### ISSUE TITLE
Request form: blocked from entering a long domain name that meets the data requirements

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2938]
### Current Behavior

On the .gov domain page, I cannot proceed with a long domain name that meets the data requirements
![Image](https://github.com/user-attachments/assets/a3d04f49-bfec-44e1-b4d2-26130f7e4de0)

### Expected Behavior

I can create a long domain name that meets the data requirements

### Steps to Reproduce

1. Navigate to the .gov domain page of the request form
2. For _What .gov domain do you want_ enter the following: rgdftgdsfsjhfkajhhrowalkjfkalsjfjaskldjkalsjdklajskldjklasdfjaksdfjaklsjdsalkas
3. Press the _Save and continue_ button

### Environment

Firefox

### Additional Context

https://github.com/user-attachments/assets/ab9973c8-a1ce-4f6a-a8c8-a9bff8fbcb80

### Issue Links

Relates to #2872 

### Prioritization Comments
**WCAG Conformance Level:**  2.1 AA
**WCAG Standard that applies:** 
3.3.1 - Error Identification (Critical)
3.3.3 - Error Suggestion (Critical)
**Critical Failure if Not Met?:**  Yes
**Prioritization:** Blocker
**Reason:** This issue affects form functionality, preventing users from submitting domain names that meet specified data requirements due to an arbitrary length restriction that has not been defined for the user. Users have to guess what length is available, which is not accessible, and places a cognitive load on all users.
